src/app/api/generateData/products/route.ts

```diff
@@ -0,0 +1,50 @@
+import { config } from "@/lib/auth";
+import { sampleData } from "@/lib/data";
+import dbConnect from "@/lib/dbConnect";
+import CounterModel from "@/lib/models/CounterModel";
+import OrderModel from "@/lib/models/OrderModel";
+import ProductModel from "@/lib/models/ProductModel";
+import UserModel from "@/lib/models/UserModel";
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+
+export const DELETE = async (request: NextRequest) => {
+  const { user } = await getServerSession(config);
+
+  if (!user || !user.isAdmin) {
+    return Response.json(
+      { message: "unauthorized" },
+      {
+        status: 401,
+      }
+    );
+  }
+
+  const { products } = sampleData;
+  await dbConnect();
+
+  await ProductModel.deleteMany();
+  await CounterModel.findByIdAndDelete("productNumber");
+  await OrderModel.deleteMany();
+
+  // Save users with incremental userNumber sequentially
+
+  // Save products with incremental productNumber sequentially
+  for (const product of products) {
+    const newProduct = new ProductModel(product);
+    await newProduct.save();
+    await new Promise((res, rej) => {
+      setTimeout(() => {
+        res("");
+      }, 10);
+    });
```

Comment:
Is this delay intentional?

Reply:
```suggestion
```

You can safely remove this code, you're already handling the update counter logic here:

https://github.com/solvd-interviews/bench-ecommerce-app-v3/blob/a2bfe913bbd6b30d25982ca6254979011526a75d/src/lib/models/ProductModel.ts#L54

Reply:
Yes Andres the delay is intentional in order to have a createdAt ordered. I remember that they bugged at the table, and this solved the problem.